packages/compass-indexes/src/components/create-index-form/partial-filter-expression.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { TextInput, CollapsibleFieldSet } from '@mongodb-js/compass-components';
+
+type PartialFilter = {
+  usePartialFilterExpression: boolean;
+  toggleUsePartialFilterExpression: (
+    usePartialFilterExpression: boolean
+  ) => any;
+  partialFilterExpression?: string;
+  partialFilterExpressionChanged: (partialFilterExpression: string) => any;
+};
+
+const PartialFilterCollapsibleFieldSet = ({
+  usePartialFilterExpression,
+  toggleUsePartialFilterExpression,
+  partialFilterExpression,
+  partialFilterExpressionChanged,
+}: PartialFilter) => {
+  return (
+    <CollapsibleFieldSet
+      toggled={usePartialFilterExpression}
+      onToggle={(checked: boolean) => toggleUsePartialFilterExpression(checked)}
+      label="Partial Filter Expression"
+      dataTestId="create-index-modal-is-pfe-checkbox"
+      description="Partial indexes only index the documents in a collection that meet a specified filter expression."
+    >
+      <TextInput
```

Comment:
Did we end up deciding if we wanted to turn this into an editor instead of a text input?

Reply:
Yeah there is already a ticket for that: https://jira.mongodb.org/browse/COMPASS-6017
This PR is too big :)